src/System.IO.FileSystem/src/System/IO/Directory.cs

```diff
@@ -268,23 +268,38 @@ public static void Move(string sourceDirName, string destDirName)
             string fulldestDirName = Path.GetFullPath(destDirName);
             string destPath = PathInternal.EnsureTrailingSeparator(fulldestDirName);
 
-            StringComparison pathComparison = PathInternal.StringComparison;
+            // Here we have a bool to check if they're equal, but with different casing
+            // We allowing moving foo to the same directory, but renaming with a different case.
+            bool sameDirectoryIgnoreCase =
```

Comment:
For clarity I would move this and the comment inline with the check where we throw for the existing directory. I'd keep the comment simpler too- "We allow moving to the same name with different casing."